packages/workbox-build/src/lib/utils/filter-files.js

```diff
@@ -1,14 +1,55 @@
+const constants = require('../constants');
+const logHelper = require('../log-helper');
+const modifyUrlPrefixTranform = require('./modify-url-prefix-transform');
+const noRevisionForUrlsMatchingTransform =
+  require('./no-revision-for-urls-matching-transform');
 const path = require('path');
 
-const logHelper = require('../log-helper');
-const constants = require('../constants');
-const modifyUrlPrefixes = require('./modify-url-prefix');
+/**
+ * A `ManifestTransform` function can be used to modify the modify the `url` or
+ * `revision` properties of some or all of the
+ * {@link module:workbox-build#ManifestEntry|ManifestEntries} in the manifest.
+ *
+ * Setting the `revision` property of an entry to a "falsey" value will cause
+ * the corresponding `url` to be precached without cache-busting parameters
+ * applied, which is to say, it implies that the URL itself contains
+ * proper versioning info.
+ *
+ * @example <caption>A transformation that prepended the origin of a CDN for any
+ * URL starting with '/assets/' could be implemented as:</caption>
+ *
+ * const cdnTransform = (manifest) => manifest.map(entry => {
+ *   const cdnOrigin = 'https://example.com';
+ *   if (entry.url.startsWith('/assets/')) {
+ *     entry.url = cdnOrigin + entry.url;
+ *   }
+ *   return entry;
+ * });
+ *
+ * @example <caption>A transformation that removes the revision field when the
+ * URL contains an 8-character hash surrounded by '.', indicating that it
+ * already contains revision information:</caption>
+ *
+ * const removeRevisionTransform = (manifest) => manifest.map(entry => {
+ *   const hashRegExp = /\.\w{8}\./;
+ *   if (entry.url.match(hashRegExp)) {
+ *     delete entry.revision;
+ *   }
+ *   return entry;
+ * });
+ *
+ * @callback ManifestTransform
+ * @param {Array<ManifestEntry>} manifest The manifest, prior to the current
+ * transformation.
+ * @return {Array<ManifestEntry>} The manifest with the transformation applied.
```

Comment:
The current implementation takes in a whole manifest (an array of one or more `ManifestEntry` objects) and returns a whole manifest.

Based on your comments, I think you're assuming that this behaves like a function that gets passed to `map()`, and which operates on one entry at a time. That's not what I ended up going with.

The function which gets passed in might use `map()` under the hood, or it might use `reduce()`, or it might just manually modify the array of `ManifestEntry` objects.

I can switch this over so that the function provided just gets used in `manifest.map(<yourFunction>)`, but there's more flexibility in this proposed implementation, and there are still checks in place to ensure that we end up with a valid manifest.

Thoughts?